services/pull/check.go

```diff
@@ -330,9 +331,22 @@ func handler(items ...string) []string {
 	return nil
 }
 
+func getPullWorkingLockKey(prID int64) string {
+	return fmt.Sprintf("pull_working_%d", prID)
+}
+
 func testPR(id int64) {
-	pullWorkingPool.CheckIn(fmt.Sprint(id))
-	defer pullWorkingPool.CheckOut(fmt.Sprint(id))
+	lock := sync_module.GetLock(getPullWorkingLockKey(id))
+	if err := lock.Lock(); err != nil {
+		log.Error("lock.Lock(): %v", err)
+		return
+	}
+	defer func() {
+		if _, err := lock.Unlock(); err != nil {
+			log.Error("lock.Unlock: %v", err)
```

Comment:
maybe we want to move the log to inside redisLockService, this "lock.Unlock" doesn't really provide information specific to this function, and then we can just `defer lock.Unlock()` here